src/modules/user/user.controller.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { DeepMocked, createMock } from '@golevelup/ts-jest';
+import {
+  BadRequestException,
+  ForbiddenException,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import getEnv from '../../config/configuration';
+import { UserController } from './user.controller';
+import { UserUseCases } from './user.usecase';
+import { NotificationService } from '../../externals/notifications/notification.service';
+import { KeyServerUseCases } from '../keyserver/key-server.usecase';
+import { CryptoService } from '../../externals/crypto/crypto.service';
+import { SharingService } from '../sharing/sharing.service';
+import { SignWithCustomDuration } from '../../middlewares/passport';
+import { newUser } from '../../../test/fixtures';
+import { AccountTokenAction } from './user.domain';
+
+jest.mock('../../config/configuration', () => {
+  return {
+    __esModule: true,
+    default: jest.fn(() => ({
+      secrets: {
+        jwt: 'Test',
+      },
+    })),
+  };
+});
+
+describe('User Controller', () => {
+  let userController: UserController;
+  let userUseCases: DeepMocked<UserUseCases>;
+  let notificationService: DeepMocked<NotificationService>;
+  let keyServerUseCases: DeepMocked<KeyServerUseCases>;
+  let cryptoService: DeepMocked<CryptoService>;
+  let sharingService: DeepMocked<SharingService>;
+
+  beforeEach(async () => {
+    userUseCases = createMock<UserUseCases>();
+    notificationService = createMock<NotificationService>();
+    keyServerUseCases = createMock<KeyServerUseCases>();
+    cryptoService = createMock<CryptoService>();
+    sharingService = createMock<SharingService>();
+
+    userController = new UserController(
+      userUseCases,
+      notificationService,
+      keyServerUseCases,
+      cryptoService,
+      sharingService,
+    );
+  });
+
+  it('should be defined', () => {
+    expect(userController).toBeDefined();
+  });
+
+  describe('POST /unblock-account', () => {
+    it('When user is not found, then returns NotFoundException', async () => {
+      userUseCases.sendAccountUnblockEmail.mockRejectedValueOnce(
+        new NotFoundException(),
+      );
+      await expect(
+        userController.requestAccountUnblock({ email: 'test@test.com' }),
+      ).rejects.toThrow(NotFoundException);
+    });
+    it('When an unexpected error is throw, then returns InternalServerException', async () => {
```

Comment:
_When an unexpected error is thrown, then the error is propagated_ could be more descriptive, especially to avoid relating legend with explicit code